CS02-React&Redux.md

```diff
@@ -0,0 +1,127 @@
+# React / Redux
+
+### 1. Describe what the project is. Does it have a mission statement?
+
+#### React is a JavaScript library for building user interfaces.
+
+
+- Declarative: React makes it painless to create interactive UIs. Design simple views for each state in your application, and React will efficiently update and render just the right components when your data changes. Declarative views make your code more predictable, simpler to understand, and easier to debug.
+
+- Component-Based: Build encapsulated components that manage their own state, then compose them to make complex UIs. Since component logic is written in JavaScript instead of templates, you can easily pass rich data through your app and keep state out of the DOM.
+
+- Learn Once, Write Anywhere: We don't make assumptions about the rest of your technology stack, so you can develop new features in React without rewriting existing code. React can also render on the server using Node and power mobile apps using React Native.
+
+#### Redux is a predictable state container for JavaScript apps.
+
+- It helps you write applications that behave consistently, run in different environments (client, server, and native), and are easy to test.
+
+### 2. Describe the history of the project -- when / how did it start? Has it grown?
+
+- React was created by Jordan Walke, a software engineer at Facebook. He was influenced by XHP, an HTML component framework for PHP. It was first deployed on Facebook's newsfeed in 2011 and later on Instagram.com in 2012. It was open-sourced at JSConf US in May 2013.'s React.js Conf in February 2015 and open-sourced in March 2015.
+
+- Redux was created by Dan Abramov and Andrew Clark in 2015. Abramov began writing the first Redux implementation while preparing for a conference talk at React Europe on hot reloading. Abramov remarks, "I was trying to make a proof of concept of Flux where I could change the logic. And it would let me time travel. And it would let me reapply the future actions on the code change."
+
+### 3. Who are the contributors to the project?
+
+- Facebook
+- Community
```

Comment:
Do you think there is any way to estimate the ratio of facebook employees to outside contributors in the contributor community?

Reply:
Yes for this thing, I think I can write a crawler to calculate the ratio.